docs/_articles/en/einstein/einstein-example-prompts.md

```diff
@@ -1,23 +1,121 @@
+title: Effective Prompt Writing
 lang: en
 ---
 
-## Generated Code Quality
+## Overview
+
+A prompt is natural language text that you craft to communicate with an AI. The response that’s generated depends on the quality of the prompt. Use prompt engineering to define the prompt and then refine it until you get a response that works for you. For E4D responses, prompt quality determines the quality of the Apex code that’s generated.
+
+## About Generated Code Quality
 
 As with all LLM-based products, the technology behind Einstein for Developers is still new. The generated output often requires editing to be customized for your individual project. Some responses aren't optimal. We'd love to hear feedback from you on the generated output through the History & Feedback Panel in the IDE. Your feedback is critical to the continuous training of our large language model (LLM) during the beta program.
 
 We strongly encourage you to let us know when you see examples of less optimal code, so that we can continue to improve generated responses. We recommend tools such as [Apex PMD](https://marketplace.visualstudio.com/items?itemName=chuckjonas.apex-pmd) (already in your Salesforce Extension Pack) and [Salesforce Code Analyzer](https://forcedotcom.github.io/sfdx-scanner/) to ensure the quality of your generated code.
 
+## Project Context and Grounding
+
+In addition to your instructions, the response generated by an AI depends on the context that your instruction is grounded with. The CodeGen2.5 model has been trained on Apex data and grounded with the context of Apex programming best practices.
+
+To extract accurate and relevant information from the model, first run **SFDX:Refresh SObjects Definitions** to gather org metadata in the form of object definitions into your project before running your query. Then use the following tips to help ground your prompts further with org context awareness to generate a high-quality response.
+
+Currently the CodeGen2.5 model can pick up on the following context from your project and prompt:
+
+- Apex code that’s present in your project in Apex class files (`.cls`).
+- Custom objects in the schema that are mentioned in the `object__c` format. Schema information takes precedence over Apex code in the current file.
+- Custom metadata when it is a part of a prompt and uses the `__mdt` suffix. The metadata added to a prompt in this way is automatically included into the context to ground generations.
+
 ## Prompt Writing Tips
 
-Prompt writing is key to quality code generation. Here are some tips for improved prompt writing and a selection of sample prompts to start you on your generative AI code-assistance journey.
+We’re listing some tips for prompt writing and also providing you with some examples that help you craft your prompts. An effective prompt consists of these four parts:
+
+1. **The Action**: The task you want the AI to accomplish.
+
+For example:
+_Create an Apex batch class named "OpenPositionMatcher"_.
```

Comment:
So from what I understand the LLM understands markdown syntax.  I think we need to include that in this doc somewhere that when the user is referring to `code` they should use back tick and if they want to include a block of code it should use ```.